lib/ansible/module_utils/azure_rm_common_ext.py

```diff
@@ -0,0 +1,190 @@
+        return self.default_compare(modifiers, new_params, old_params, '', self.results)
+
+    def create_compare_modifiers(self, arg_spec, path, result):
+        for k in arg_spec.keys():
+            o = arg_spec[k]
+            updatable = o.get('updatable', True)
+            comparison = o.get('comparison', 'default')
+            p = (path +
+                 ('/' if len(path) > 0 else '') +
+                 o.get('disposition', '*').replace('*', k) +
+                 ('/*' if o['type'] == 'list' else ''))
+            if comparison != 'default' or not updatable:
+                result[p] = {'updatable': updatable, 'comparison': comparison}
+            if o.get('options'):
+                self.create_compare_modifiers(o.get('options'), p, result)
+
+    def default_compare(self, modifiers, new, old, path, result):
+        '''
+            Default dictionary comparison.
+            This function will work well with most of the Azure resources.
+            It correctly handles "location" comparison.
+
+            Value handling:
+                - if "new" value is None, it will be taken from "old" dictionary if "incremental_update"
+                  is enabled.
+            List handling:
+                - if list contains "name" field it will be sorted by "name" before comparison is done.
+                - if module has "incremental_update" set, items missing in the new list will be copied
+                  from the old list
+
+            Warnings:
+                If field is marked as non-updatable, appropriate warning will be printed out and
+                "new" structure will be updated to old value.
+
+            :modifiers: Optional dictionary of modifiers, where key is the path and value is dict of modifiers
+            :param new: New version
+            :param old: Old version
+
+            Returns True if no difference between structures has been detected.
+            Returns False if difference was detected.
+        '''
+        if new is None:
+            return True
+        elif isinstance(new, dict):
+            if not isinstance(old, dict):
+                result['compare'] = 'changed [' + path + '] old dict is null'
+                return False
+            for k in new.keys():
+                if not self.default_compare(modifiers, new.get(k), old.get(k, None), path + '/' + k, result):
+                    return False
+            return True
+        elif isinstance(new, list):
+            if not isinstance(old, list) or len(new) != len(old):
+                result['compare'] = 'changed [' + path + '] length is different or null'
+                return False
+            if isinstance(old[0], dict):
+                key = None
+                if 'id' in old[0] and 'id' in new[0]:
+                    key = 'id'
+                elif 'name' in old[0] and 'name' in new[0]:
+                    key = 'name'
+                else:
+                    key = list(old[0].keys())[0]
```

Comment:
This is more efficient:
```suggestion
                    key = next(iter(old[0])
```

Reply:
thanks!